### PR TITLE
Kured

### PR DESCRIPTION
**Pull request by irish1986**
# Proposed Changes

Following PR #191 , I am proposing new feature to be added, although this slightly customize this project adding a none essentially feature it is very useful mechanism.

I proposed to add [kured](https://github.com/kubereboot/kured) workload as part of the default configuration.  kured watches for the presence of a reboot sentinel i.e.: `/var/run/reboot-required` and allows to configure how nodes should be reboot as desired.  The [kured docs](https://kured.dev/) is extensive and allows all sort of complexe use case but essentially it a usefule way to control how nodes reboots without creating too much chaos and potential downtime.

For instance, kured can automated cordon and drains worker nodes before reboot and uncordoning them after which typically requires some `kubectl-fu ninja` skill. kured automated all that as per the configuration rules set.

- Add kured manifest
- I have not added any post configuration since each use case is unique.

Finally, there are plenty of cool ways to integrate kured into prometheus, slack and other means of notification since again this is individual to each installation.

## Checklist

- [X] Tested locally
- [X] Ran `site.yml` playbook
- [X] Ran `reset.yml` playbook
- [ ] Did not add any unnecessary changes
- [X] Ran pre-commit install at least once before committing
- [X] 🚀

### CODE REVIEW

File: roles/k3s/master/tasks/main.yml [hunk 1]
```diff
@@ -13,51 +13,11 @@
   args:
     warn: false  # The ansible systemd module does not support reset-failed
```
Comment (sleiner) [inline roles/k3s/master/tasks/main.yml:15]: Do we need to include the tasks from `kured.yml` here?

File: inventory/sample/group_vars/all.yml [hunk 2]
```diff
@@ -43,6 +43,9 @@ extra_agent_args: >-
 # image tag for kube-vip
 kube_vip_tag_version: "v0.5.7"
 
+# image tag for kured
+kured_tag_version: "1.12.0"
```
Comment (sleiner) [inline inventory/sample/group_vars/all.yml:47]: Can/should we make the installation of kured optional?

Comment (irish1986) [reply inventory/sample/group_vars/all.yml:47]: > Yes we could, I am thinking adding a when  statement **kured_tag_version** is defined to include the task in the playbook instead managing two separate variables. How would that be for an implementation?

File: roles/k3s/master/tasks/main.yml (hunk 1 above)
Comment (irish1986) [reply roles/k3s/master/tasks/main.yml]: I did messed up there, I think working late in December had my brain fuzzy and forgotten about the important task.

**Comment by timothystewart6** [comment]
see comment https://github.com/techno-tim/k3s-ansible/pull/192#issuecomment-1376588631

**Pull request closed without merging**